internal/operators/oai/oai_operator.go

```diff
@@ -0,0 +1,263 @@
+package oai
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/go-version"
+	"github.com/kelseyhightower/envconfig"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/internal/operators/api"
+	"github.com/openshift/assisted-service/models"
+	"github.com/openshift/assisted-service/pkg/conversions"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	operatorName      = "oai"
+	operatorNamespace = "redhat-ods-operator"
+	operatorFullName  = "OpenShift AI"
+	subscriptionName  = "rhods-operator"
+)
+
+// operator is an OpenShift AI OLM operator plugin.
+type operator struct {
+	log    logrus.FieldLogger
+	config *Config
+}
+
+// NewOperator creates new OpenShift AI operator.
+func NewOperator(log logrus.FieldLogger) *operator {
+	config := &Config{}
+	err := envconfig.Process(common.EnvConfigPrefix, config)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+	return &operator{
+		log:    log,
+		config: config,
+	}
+}
+
+// GetName reports the name of an operator.
+func (o *operator) GetName() string {
+	return operatorName
+}
+
+func (o *operator) GetFullName() string {
+	return operatorFullName
+}
+
+// GetDependencies provides a list of dependencies of the Operator
+func (o *operator) GetDependencies(c *common.Cluster) ([]string, error) {
+	return make([]string, 0), nil
+}
+
+// GetClusterValidationID returns cluster validation ID for the operator.
+func (o *operator) GetClusterValidationID() string {
+	return string(models.ClusterValidationIDOAIRequirementsSatisfied)
+}
+
+// GetHostValidationID returns host validation ID for the operator.
+func (o *operator) GetHostValidationID() string {
+	return string(models.HostValidationIDOAIRequirementsSatisfied)
+}
+
+// ValidateCluster checks if the cluster satisfies the requirements to install the operator.
+func (o *operator) ValidateCluster(ctx context.Context, cluster *common.Cluster) (result api.ValidationResult,
+	err error) {
+	// Check the OpenShift version:
+	curVersion, err := version.NewVersion(cluster.OpenshiftVersion)
+	if err != nil {
+		result.Status = api.Failure
+		result.ValidationId = o.GetHostValidationID()
+		result.Reasons = []string{
+			err.Error(),
+		}
+		return
+	}
+	minVersion, err := version.NewVersion(o.config.MinOpenShiftVersion)
```

Comment:
Why do we need validate here and in `featureSupport` as well ?

Reply:
I don't have a good answer for that, will check.

Reply:
You are right @danmanor: this isn't necessary. The same check is performed when the cluster is created, using the feature support mechanism. I will remove this in the next version of the pull request.

Reply:
Removed this.